lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1535,7 +1535,18 @@ namespace {
         if (Name.str() == "id" && Impl.SwiftContext.LangOpts.EnableIdAsAny) {
           Impl.SpecialTypedefNames[Decl->getCanonicalDecl()] =
               MappedTypeNameKind::DoNothing;
-          return Impl.SwiftContext.getAnyDecl();
+
+          auto DC = Impl.importDeclContextOf(Decl, importedName.EffectiveContext);
+          if (!DC) return nullptr;
+
+          // create arbitrary Decl for the implicit Any type
+          return Impl.createDeclWithClangNode<TypeAliasDecl>(
```

Comment:
I want to conditionalize the mapping based on `LangOpts` while staging in id-as-Any, and it wasn't immediately obvious how to do that with the existing mapping path, which expands the `MappedTypes.def` xmacro with an entry to map `id` to `AnyObject`. How does this leak `id` into the public interface? AFAICT this doesn't import the Clang decl, it just puts it in `SpecialTypedefNames` mapping it to `DoNothing`, following what the `MappedTypes` expansion does.